semantic-core/src/Data/Core.hs

```diff
@@ -215,27 +216,27 @@ a .= b = send (a := b)
 infix 3 .=
 
 
-data Ann f a
-  = Ann Loc (f a)
+data Ann ann f a
+  = Ann ann (f a)
```

Comment:
Generalizing the annotation type is a step towards extracting it right out of `semantic-core` altogether. Eventually this, and most of the other syntax machinery, will live in a package like `fused-syntax` instead.